front-end/src/App.js

```diff
@@ -1,30 +1,46 @@
-import { React, useState } from 'react';
+import { React, useState, useEffect } from 'react';
 import { Routes, Route } from "react-router-dom"
 import Dashboard from './component/Dashboard';
 import Register from './component/user/Register';
 import Login from './component/user/Login';
 import Profile from './component/user/Profile'
+import Inventory from './component/management/Inventory';
 import "./styles/App.css"
 
 const App = () => {
-  const [userInformation, setUserInformation] = useState({
-    username: null,
-    email: null,
-    id: null
-  })
+  const [userInformation, setUserInformation] = useState(() => {
+    const storedUserInformation = sessionStorage.getItem('userInformation');
+    return storedUserInformation ? JSON.parse(storedUserInformation) : {
+      username: null,
+      email: null,
+      id: null
+    };//retrieves the user information if it exists. If it doesnt return a null/empty user information
+  });
+
+  useEffect(() => {
+    /* Update sessionStorage whenever userInformation changes*/
+    sessionStorage.setItem('userInformation', JSON.stringify(userInformation));
+  }, [userInformation]);
+
+  const [inventory, setUserInventory] = useState([{
+    id: null,
+    ingredient: null,
+    quantity: null
+  }]);//used to set and get the usersInventory
 
```

Comment:
Why do we need to define the inventory info in App.js? It makes sense for userInfo because we are passing it to all other components, but the inventory info is only needed for the inventory component. And if we need to get the inventory in another component thats what passing the userInfo is for